src/network/backbox/rest/custom/api.pm

```diff
@@ -0,0 +1,265 @@
+        $self->{output}->option_exit();
+    }
+
+    return 0;
+}
+
+sub get_connection_infos {
+    my ($self, %options) = @_;
+
+    return $self->{hostname} . '_' . $self->{http}->get_port();
+}
+
+sub get_hostname {
+    my ($self, %options) = @_;
+
+    return $self->{hostname};
+}
+
+sub get_port {
+    my ($self, %options) = @_;
+
+    return $self->{port};
+}
+
+sub json_decode {
+    my ($self, %options) = @_;
+
+    $options{content} =~ s/\r//mg;
+    my $decoded;
+    eval {
+        $decoded = JSON::XS->new->utf8->decode($options{content});
+    };
+    if ($@) {
+        $self->{output}->add_option_msg(short_msg => "Cannot decode json response: $@");
+        $self->{output}->option_exit();
+    }
+
+    return $decoded;
+}
+
+sub build_options_for_httplib {
+    my ($self, %options) = @_;
+
+    $self->{option_results}->{hostname} = $self->{hostname};
+    $self->{option_results}->{port}     = $self->{port};
+    $self->{option_results}->{proto}    = $self->{proto};
+    $self->{option_results}->{timeout}  = $self->{timeout};
+}
+
+sub settings {
+    my ($self, %options) = @_;
+
+    $self->build_options_for_httplib();
+    $self->{http}->add_header(key => 'Accept', value => 'application/json');
+    $self->{http}->set_options(%{$self->{option_results}});
+}
+
+sub request {
+    my ($self, %options) = @_;
+
+    my $endpoint = $options{full_endpoint};
+    if (!defined($endpoint)) {
+        $endpoint = $self->{url_path} . $options{endpoint};
+    }
+
+    $self->settings();
+
+    my $content = $self->{http}->request(
+        method          => 'GET',
+        url_path        => $endpoint,
+        get_param       => $options{get_param},
+        header          => [
+            'AUTH: ' . $self->{api_token}
+        ],
+        warning_status  => '',
+        unknown_status  => '',
+        critical_status => ''
+    );
+
+    # Maybe there is an issue with the token. So we retry.
+    if ($self->{http}->get_code() < 200 || $self->{http}->get_code() >= 300) {
+        $content = $self->{http}->request(
+            url_path        => $endpoint,
+            get_param       => $options{get_param},
+            header          => [
+                'AUTH: ' . $self->{api_token}
+            ],
+            unknown_status  => $self->{unknown_http_status},
+            warning_status  => $self->{warning_http_status},
+            critical_status => $self->{critical_http_status}
+        );
+    }
+
+    my $decoded = $self->json_decode(content => $content);
+    if (!defined($decoded)) {
+        $self->{output}->add_option_msg(short_msg => 'Error while retrieving data (add --debug option for detailed message)');
+        $self->{output}->option_exit();
+    }
+
+    return $decoded;
+}
+
+sub get_backup_jobs_status {
+    my ($self, %options) = @_;
+
+    my $endpoint = 'dashboard/backupStatus';
+    if (defined($options{filter_type}) && $options{filter_type} ne '') {
```

Comment:
You could have used `if ( !centreon::plugins::misc::is_empty($options{filter_type}) )` instead